### PR TITLE
Add support for Managed zone resource of Integration Connectors.

### PR DESCRIPTION
**Pull request by balanaguharsha**
**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_integration_connectors_managed_zone`
```

### CODE REVIEW

File: mmv1/products/integrationconnectors/ManagedZone.yaml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  insert_minutes: 5
+  update_minutes: 5
+  delete_minutes: 5
+create_url: projects/{{project}}/locations/global/managedZones?managedZoneId={{name}}
+update_verb: :PATCH
+update_mask: true
+autogen_async: true
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "integration_connectors_connection_basic"
```
Comment (trodge) [inline mmv1/products/integrationconnectors/ManagedZone.yaml:36]: Should this be `integration_connectors_managed_zone`?

**Comment by balanaguharsha** [comment]
> Hi there, I'm the Modular magician. I've detected the following information about your changes:
> 
> ## Diff report
> Your PR generated some diffs in downstreams - here they are.
> 
> `google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10580-old..auto-pr-10580) ( 6 files changed, 1250 insertions(+), 2 deletions(-)) `google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10580-old..auto-pr-10580) ( 6 files changed, 1250 insertions(+), 2 deletions(-)) `terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-10580-old..auto-pr-10580) ( 1 file changed, 116 insertions(+))
> 
> ## Missing test report
> Your PR includes resource fields which are not covered by any test.
> 
> Resource: `google_integration_connectors_managed_zone` (3 total tests) Please add an acceptance test which includes these fields. The test should include the following:
> 
> ```terraform
> resource "google_integration_connectors_managed_zone" "primary" {
>   project = # value needed
> }
> ```

I am not sure how to solve this, please suggest, shouldn't project be from the context anyways, the resource doesn't have a field called project explicitly defined in the resource's yaml

**Comment by balanaguharsha** [comment]
PTAL at convenience.

**Comment by trodge** [comment]
> > Hi there, I'm the Modular magician. I've detected the following information about your changes:
> > ## Diff report
> > Your PR generated some diffs in downstreams - here they are.
> > `google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10580-old..auto-pr-10580) ( 6 files changed, 1250 insertions(+), 2 deletions(-)) `google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10580-old..auto-pr-10580) ( 6 files changed, 1250 insertions(+), 2 deletions(-)) `terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-10580-old..auto-pr-10580) ( 1 file changed, 116 insertions(+))
> > ## Missing test report
> > Your PR includes resource fields which are not covered by any test.
> > Resource: `google_integration_connectors_managed_zone` (3 total tests) Please add an acceptance test which includes these fields. The test should include the following:
> > ```terraform
> > resource "google_integration_connectors_managed_zone" "primary" {
> >   project = # value needed
> > }
> > ```
> 
> I am not sure how to solve this, please suggest, shouldn't project be from the context anyways, the resource doesn't have a field called project explicitly defined in the resource's yaml

This can be ignored.

File: mmv1/products/integrationconnectors/ManagedZone.yaml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  - !ruby/object:Api::Type::String
+    name: "name"
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      Name of Managed Zone needs to be created.
+
+properties:
+  - !ruby/object:Api::Type::Time
```
Comment (trodge) [inline mmv1/products/integrationconnectors/ManagedZone.yaml:77]: We usually omit create time and update time fields from new resources. Is there a reason to include them here?

**Review by trodge — CHANGES_REQUESTED** [review]
LGTM, just one nit.

**Comment by balanaguharsha** [comment]
Have resolved your other comment.

**Comment by balanaguharsha** [comment]
PTAL

**Review by trodge — APPROVED** [review]

**Pull request merged**